cinema/urls.py

```diff
@@ -1,10 +1,38 @@
-from django.urls import path
+from django.urls import path, include
 
-from cinema.views import movie_list, movie_detail
+from rest_framework import routers
 
-urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
-]
+from cinema.views import (
+    MovieViewSet,
+    GenreList,
+    GenreDetail,
+    ActorList,
+    ActorDetail,
+    CinemaHallViewSet
+)
 
 app_name = "cinema"
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet, basename="movies")
+router.register("cinema_halls", CinemaHallViewSet, basename="cinemahall")
+
+
+urlpatterns = [
+    path("", include(router.urls)),
+    path("genres/", GenreList.as_view(), name="genre-list"),
+    path("genres/create/", GenreList.as_view(), name="genre-create"),
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genre-detail"),
+    path(
+        "genres/<int:pk>/update/",
+        GenreDetail.as_view(),
+        name="genre-update"
+    ),
+    path(
+        "genres/<int:pk>/delete/",
+        GenreDetail.as_view(),
+        name="genre-delete"
+    ),
+    path("actors/", ActorList.as_view(), name="actor-list"),
+    path("actors/", ActorList.as_view(), name="actor-list"),
```

Comment:
It seems like there is a duplicate path for 'actors/'. You have two identical paths with the same name 'actor-list'. Consider removing one of them to avoid redundancy.